File: python/coding/loop/str2ascii.py
#!/usr/bin/env python3

import sys


def str2ascii(message):
    """convert message to ascii"""
    for c in message:
        print(f"|{ord(c):3}|{c:1}|", sep="|")


def main(message):
    # Classic way
    str2ascii(message)

    print()

    # Python style
    print(*[ord(c) for c in list(message)])


# ------------------------------------------------------------------------------
if __name__ == "__main__":
    if len(sys.argv) != 2:
        print(f"Usage: {sys.argv[0]} <string>")
    else:
        main(sys.argv[1])

File: ruby/coding/loop/str2ascii.rb
#!/usr/bin/env ruby

def convert_to_ascii msg
  msg_to_lst = msg.split('')
  msg_to_lst.each do |c|
    puts("|#{c.ord.to_s.ljust(3)}|#{c}|")
    # c.ord -> integer
    # <String>.ljust(<integer>[...]) -> new_str
  end
end

def main
  convert_to_ascii ARGV[0]
end

if __FILE__ == $0
  main
end
